include/seqan3/io/alignment_file/format_base.hpp

```diff
@@ -0,0 +1,724 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides the seqan3::format_sam_base that can be inherited from.
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <iterator>
+#include <string>
+#include <vector>
+
+#include <seqan3/core/char_operations/predicate.hpp>
+#include <seqan3/core/concept/core_language.hpp>
+#include <seqan3/core/concept/tuple.hpp>
+#include <seqan3/core/detail/reflection.hpp>
+#include <seqan3/core/detail/to_string.hpp>
+#include <seqan3/core/type_traits/range.hpp>
+#include <seqan3/core/type_traits/template_inspection.hpp>
+#include <seqan3/io/alignment_file/detail.hpp>
+#include <seqan3/io/alignment_file/header.hpp>
+#include <seqan3/io/alignment_file/input_format_concept.hpp>
+#include <seqan3/io/alignment_file/input_options.hpp>
+#include <seqan3/io/alignment_file/output_format_concept.hpp>
+#include <seqan3/io/alignment_file/output_options.hpp>
+#include <seqan3/io/alignment_file/sam_tag_dictionary.hpp>
+#include <seqan3/io/detail/ignore_output_iterator.hpp>
+#include <seqan3/io/detail/misc.hpp>
+#include <seqan3/io/stream/iterator.hpp>
+#include <seqan3/io/sequence_file/output_options.hpp>
+#include <seqan3/range/detail/misc.hpp>
+#include <seqan3/range/views/char_to.hpp>
+#include <seqan3/range/views/istreambuf.hpp>
+#include <seqan3/range/views/repeat_n.hpp>
+#include <seqan3/range/views/slice.hpp>
+#include <seqan3/range/views/take_until.hpp>
+#include <seqan3/range/views/to_char.hpp>
+#include <seqan3/range/views/to.hpp>
+#include <seqan3/range/views/zip.hpp>
+#include <seqan3/std/algorithm>
+#include <seqan3/std/charconv>
+#include <seqan3/std/concepts>
+#include <seqan3/std/ranges>
+#include <seqan3/version.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief       The alignment base format.
+ * \implements  AlignmentFileFormat
+ * \ingroup     alignment_file
+ *
+ * \details
+ *
+ * Since the SAM and BAM format share a lot of functionality, this abstract base class defines common member variables
+ * and functions that are used in both formats.
+ */
+struct format_sam_base
+{
+public:
+    //!\brief The format version string.
+    static constexpr char format_version[4] = "1.6";
+
+protected:
+    //!\brief A buffer used when parsing arithmetic values with std::from_chars.
+    std::array<char, 316> buffer{}; // Doubles can be up to 316 characters
```

Comment:
```suggestion
    std::array<char, 316> arithmetic_buffer{}; // Doubles can be up to 316 characters
```

Since you have other buffers, as well?